console/src/state/async_ops.rs

```diff
@@ -0,0 +1,308 @@
+use crate::{
```

Comment:
this file looks fine overall, but also, it does occur to me that we're really starting to reach the point where there's a _lot_ of very similar boilerplate for different parts of the data model (tasks, resources, async ops...). it would maybe be worth thinking about whether some of this code, like the sorting by fields, could be made generic so we don't have to write identical boilerplate quite so many times.

but, that's not really a problem for this branch, just something we might want to think about addressing later...